src/HotChocolate/MongoDb/test/Data.MongoDb.Integration.Tests/AnnotationBasedIntegrationTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿using CookieCrumble;
+using HotChocolate.Data.MongoDb.Integration.Tests.Models;
+using HotChocolate.Data.MongoDb.Integration.Tests.Schema;
+using HotChocolate.Execution;
+using Microsoft.Extensions.DependencyInjection;
+using MongoDB.Driver;
+using Squadron;
+using CreateCollectionOptions = Squadron.CreateCollectionOptions;
+
+namespace HotChocolate.Data.MongoDb.Integration.Tests;
+
+[Collection("Database")]
+public class AnnotationBasedIntegrationTests : IClassFixture<MongoResource>
+{
+    private readonly MongoResource _mongoDb;
+
+    public AnnotationBasedIntegrationTests(MongoResource mongoResource)
+    {
+        _mongoDb = mongoResource;
+    }
+
+    [Fact]
+    public async Task MoviesSchemaIntegrationTests()
+    {
+        // arrange
+        var tester = await CreateSchema();
+
+        // act
+        var res1 = await tester.ExecuteAsync(@"{
```

Comment:
@PascalSenn this sorting problem started when I tried to fix the Neo4J tests which produce indeterminate ordering on nested collections. This test was designed to prove the existing data layer cannot handle ordering on a nested collection as part of an integration test. This test matches an equivalent test in the Neo4J integration project.

I think the MongoDb and Neo4J pipelines should be able to defer to the queryable paging and sorting providers. so that in memory paging and sorting can be applied to collections.

Thoughts? :D